packages/cache/src/internal/cacheHttpClient.ts

```diff
@@ -113,6 +114,8 @@ export async function getCacheEntry(
   const cacheResult = response.result
   const cacheDownloadUrl = cacheResult?.archiveLocation
   if (!cacheDownloadUrl) {
+    // List cache for primary key only if cache miss occurs
```

Comment:
This method exists in line 108 if the status code is 204. 
In which case will there be cacheResult without archiveLocation?

Reply:
https://github.com/actions/toolkit/blob/main/packages/cache/src/cache.ts#L96
It checks after returning from here.

Reply:
rephrasing the question:
If there is a cache miss, then the method exists in line 108 and this line is not reachable. Is that expected?

Reply:
In case of cache miss condition at line 107 is false as statusCode wouldn't be 204. Then it moves ahead to check if status is not success. Then if it is success but not 204 then it moves ahead to check more.

Reply:
I have moved the logic up to line 108 when it returns null for cache miss